frontend/src/container/NewWidget/LeftContainer/QuerySection/index.tsx

```diff
@@ -168,51 +170,63 @@ function QuerySection({
 		},
 	];
 
-	const items = [
-		{
-			key: EQueryType.QUERY_BUILDER,
-			label: (
-				<Button className="nav-btns">
-					<Atom size={14} />
-					<Typography>Query Builder</Typography>
-				</Button>
-			),
-			tab: <Typography>Query Builder</Typography>,
-			children: (
-				<QueryBuilder
-					panelType={selectedGraph}
-					filterConfigs={filterConfigs}
-					version={selectedDashboard?.data?.version || 'v3'}
-				/>
-			),
-		},
-		{
-			key: EQueryType.CLICKHOUSE,
-			label: (
-				<Button className="nav-btns">
-					<Terminal size={14} />
-					<Typography>ClickHouse Query</Typography>
-				</Button>
-			),
-			tab: <Typography>ClickHouse Query</Typography>,
-			children: <ClickHouseQueryContainer />,
-		},
-		{
-			key: EQueryType.PROM,
-			label: (
-				<Tooltip title="PromQL">
+	const items = useMemo(() => {
+		const baseItems = [
+			{
+				key: EQueryType.QUERY_BUILDER,
+				label: (
 					<Button className="nav-btns">
-						<PromQLIcon
-							fillColor={isDarkMode ? Color.BG_VANILLA_200 : Color.BG_INK_300}
-						/>
-						<Typography>PromQL</Typography>
+						<Atom size={14} />
+						<Typography>Query Builder</Typography>
 					</Button>
-				</Tooltip>
-			),
-			tab: <Typography>PromQL</Typography>,
-			children: <PromQLQueryContainer />,
-		},
-	];
+				),
+				tab: <Typography>Query Builder</Typography>,
+				children: (
+					<QueryBuilder
+						panelType={selectedGraph}
+						filterConfigs={filterConfigs}
+						version={selectedDashboard?.data?.version || 'v3'}
+					/>
+				),
+			},
+			{
+				key: EQueryType.CLICKHOUSE,
+				label: (
+					<Button className="nav-btns">
+						<Terminal size={14} />
+						<Typography>ClickHouse Query</Typography>
+					</Button>
+				),
+				tab: <Typography>ClickHouse Query</Typography>,
+				children: <ClickHouseQueryContainer />,
+			},
+		];
+
+		if (selectedGraph !== PANEL_TYPES.TABLE) {
```

Comment:
NIT: better to maintain a map of supported types vs panel types , easy to read and scalable to maintain